packages/react/src/auth/useSignInWithEmailAndPasswordMutation.test.tsx

```diff
@@ -0,0 +1,185 @@
+import React from "react";
+import {
+  describe,
+  expect,
+  test,
+  beforeEach,
+  afterEach,
+  vi,
+  type MockInstance,
+} from "vitest";
+import { renderHook, act, waitFor } from "@testing-library/react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { useSignInWithEmailAndPasswordMutation } from "./useSignInWithEmailAndPasswordMutation";
+import { auth, wipeAuth } from "~/testing-utils";
+import { createUserWithEmailAndPassword } from "firebase/auth";
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: { retry: false },
+    mutations: { retry: false },
+  },
+});
+
+const wrapper = ({ children }: { children: React.ReactNode }) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+);
+
+describe("useSignInWithEmailAndPasswordMutation", () => {
+  beforeEach(async () => {
+    queryClient.clear();
+    await wipeAuth();
+  });
+
+  afterEach(async () => {
+    await auth.signOut();
+  });
+
+  test("successfully signs in with email and password", async () => {
+    const email = "tqf@invertase.io";
+    const password = "tanstackQueryFirebase#123";
+    await createUserWithEmailAndPassword(auth, email, password);
+
+    const { result } = renderHook(
+      () => useSignInWithEmailAndPasswordMutation(auth, email, password),
+      { wrapper }
+    );
+
+    await act(async () => result.current.mutate());
+
+    await waitFor(async () => expect(result.current.isSuccess).toBe(true));
+
+    expect(result.current.data?.user.email).toBe(email);
+  });
+
+  test("fails to sign in with incorrect password", async () => {
+    const email = "tqf@invertase.io";
+    const password = "tanstackQueryFirebase#123";
+    const wrongPassword = "wrongpassword";
+
+    await createUserWithEmailAndPassword(auth, email, password);
+
+    const { result } = renderHook(
+      () => useSignInWithEmailAndPasswordMutation(auth, email, wrongPassword),
+      { wrapper }
+    );
+
+    await act(async () => {
+      result.current.mutate();
+    });
+
+    await waitFor(() => expect(result.current.isError).toBe(true));
+
+    expect(result.current.error).toBeDefined();
+    expect(result.current.isSuccess).toBe(false);
+    // TODO: Assert Firebase error for auth/wrong-password
+  });
+
+  test("fails to sign in with non-existent email", async () => {
+    const email = "tqf@invertase.io";
+    const password = "tanstackQueryFirebase#123";
+
+    const { result } = renderHook(
+      () => useSignInWithEmailAndPasswordMutation(auth, email, password),
+      { wrapper }
+    );
+
+    await act(async () => {
+      result.current.mutate();
+    });
+
+    await waitFor(() => expect(result.current.isError).toBe(true));
+
+    expect(result.current.error).toBeDefined();
+    expect(result.current.isSuccess).toBe(false);
+    // TODO: Assert Firebase error for auth/user-not-found
```

Comment:
the assertion for `expectFirebaseError` that covers scope beyond `expectFirestoreError` was implemented via https://github.com/invertase/react-query-firebase/pull/114 so was thinking so was thinking will rebase with https://github.com/invertase/react-query-firebase/pull/114 once its merged.